### PR TITLE
PMM-4834 Upload SSH key.

### PR DESCRIPTION
**Pull request by lunaticusgreen**
https://github.com/Percona-Lab/pmm-submodules/pull/559

### CODE REVIEW

**Comment by lunaticusgreen** [comment]
https://github.com/Percona-Lab/pmm-submodules/pull/559

**Comment by lunaticusgreen** [comment]
image
image
image

File: pmm-app/src/pmm-settings/Parts/UploadSSH/__snapshots__/UploadSSH.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          onChange={[Function]}
+          onFocus={[Function]}
+          placeholder="Enter ssh key"
+          rows={5}
+          style={
+            Object {
+              "width": "100%",
+            }
+          }
+          value=""
```
Comment (BupycHuk) [inline pmm-app/src/pmm-settings/Parts/UploadSSH/__snapshots__/UploadSSH.test.tsx.snap]: I expected to see `value="test_ssh_key"` here

File: pmm-app/src/pmm-settings/Parts/Settings/Settings.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+const SettingsPart = props => {
+  const [loading, setLoading] = useState(false);
+  const { settings } = props;
+  const onSubmit = async values => {
+    const updatedSettings: SettingsInterface = {
+      data_retention: values.data_retention_count + values.data_retention_units,
+      metrics_resolutions: dataRetentionValues[values.metrics_resolutions_slider],
+    };
+
+    if (values.telemetry_enabled) {
```
[the remark this answers is not in the document]
Comment (lunaticusgreen) [reply pmm-app/src/pmm-settings/Parts/Settings/Settings.tsx]: Nice one, will use it!

File: pmm-app/src/pmm-settings/Parts/AlertManager/__snapshots__/AlertManager.test.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          onChange={[Function]}
+          onFocus={[Function]}
+          placeholder="Enter IP"
+          style={
+            Object {
+              "width": "100%",
+            }
+          }
+          type="text"
+          value=""
```
[the remark this answers is not in the document]
Comment (lunaticusgreen) [reply pmm-app/src/pmm-settings/Parts/AlertManager/__snapshots__/AlertManager.test.tsx.snap]: Can't tell you that about input fields, need to do more extensive research, maybe it's related to react final form, or maybe to jest itself. I'll work on this in future tasks tests

File: pmm-app/src/react-plugins-deps/components/LatencyChart/LatencyChart.tsx [hunk 4]
```diff
@@ -1,111 +1,111 @@
...
+//         .attr('cx', x(p99) + '')
+//         .attr('cy', '13px');
+//     }
+//   }
+//   render() {
+//     return <div className={'d3-bar-chart-container'} ref={'graphContainer'}></div>;
+//   }
+// }
+//
+// export default LatencyChart;
```
Comment (AlekSi) [inline pmm-app/src/react-plugins-deps/components/LatencyChart/LatencyChart.tsx]: Is it commented out completely?

Comment (lunaticusgreen) [reply pmm-app/src/react-plugins-deps/components/LatencyChart/LatencyChart.tsx]: No, will be used here https://github.com/percona/grafana-dashboards/pull/542, it's already uncommented

**Pull request merged**